bundles/admin/admin-announcements/view/AnnouncementsForm.jsx

```diff
@@ -0,0 +1,163 @@
+
+  const onFinish  = fieldsValue => {
+    // Should format date value before submit.
+    const rangeValue = fieldsValue["range_picker"];
+
+    const values = {
+      title: fieldsValue["title"],
+      content: fieldsValue["content"],
+      begin_date: rangeValue[0].format(DATEFORMAT), 
+      end_date: rangeValue[1].format(DATEFORMAT),
+      active: fieldsValue["active"]
+    };
+
+    if (form.id === undefined) {
+      controller.saveAnnouncement(values);
+    } else {
+      values.id = form.id;
+      controller.updateAnnouncement(values);
+    }
+  }
+  
+  //Set initial values to date range depending on if we are editing or creating an announcement
+  const rangeInitial = () => {
+
+    if (form.begin_date && form.end_date) {
+      return [moment(form.begin_date, DATEFORMAT), moment(form.end_date, DATEFORMAT)]; 
+    } else {
+      return [moment(moment(),DATEFORMAT), moment(moment(),DATEFORMAT)];  
+    }
+
+  } 
+
+  //Active value set depending on if creating a new announcement or editing an old one
+  const activeInitial = () => {
+    if(form.active === undefined) {
+      return true;
+    } else {
+      return form.active;
+    }
+  }
+
+    return (
+      <div>
+            <Form layout="vertical" 
+              onFinish={onFinish} 
+              initialValues={{
+                title: form.title !== undefined ? form.title : title,
+                content: form.content,
+                range_picker: rangeInitial(),
+                active: activeInitial(),
+              }}>
+              <Form.Item
+                name="title"
+                label={<Message messageKey='newAnnouncement.title' />}
+                rules={[
+                  {
+                    required: true,
+                    message: <Message messageKey='titleError' />,
+                    whitespace: true
+                  }
+                ]}
+                value={title}
+                onChange={ evt => controller.updateTitle(evt.target.value)}
+              >
+                <Input />
+              </Form.Item>
+              <Form.Item
+                name="content"
+                label={<Message messageKey='newAnnouncement.content' />}
+                rules={[
+                  {
+                    required: true,
+                    message: <Message messageKey='contentError' />,
+                    whitespace: true
+                  }
+                ]}
+              >
+                <TextArea rows={4}/>
+              </Form.Item>
+              <Form.Item
+                name="range_picker"
+                label={<Message messageKey='newAnnouncement.date-range' />}
+                {...rangeConfig}
+              >
+                <DateRange popupStyle={{zIndex: '999999'}} />
+              </Form.Item>
+              <Form.Item name="active" label={<Message messageKey='newAnnouncement.show-popup' />} valuePropName="checked">
+                <Switch onClick={() => controller.toggleActive()}/>
```

Comment:
Same here. With antd form you can just handle the onFinish() and not call controller from fields.